pkg/policy/background.go

```diff
@@ -3,142 +3,74 @@ package policy
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
-	gojmespath "github.com/jmespath/go-jmespath"
 	kyverno "github.com/kyverno/kyverno/api/kyverno/v1"
 	"github.com/kyverno/kyverno/pkg/engine/context"
 	"github.com/kyverno/kyverno/pkg/engine/variables"
-	"github.com/kyverno/kyverno/pkg/utils"
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-//ContainsVariablesOtherThanObject returns error if variable that does not start from request.object
-func ContainsVariablesOtherThanObject(policy kyverno.ClusterPolicy) error {
-	var err error
-	for idx, rule := range policy.Spec.Rules {
-		if path := userInfoDefined(rule.MatchResources.UserInfo); path != "" {
-			return fmt.Errorf("invalid variable used at path: spec/rules[%d]/match/%s", idx, path)
+//ContainsUserVariables returns error if variable that does not start from request.object
+func containsUserVariables(policy *kyverno.ClusterPolicy, vars [][]string) error {
```

Comment:
What about `request.operation`, `request.namespace` etc?

Reply:
Those are covered in the prior check - `hasInvalidVariables`:


```golang

func ValidateVariables(p *kyverno.ClusterPolicy, backgroundMode bool) error {
	vars := hasVariables(p)
	if len(vars) == 0 {
		return nil
	}

	if err := hasInvalidVariables(p, backgroundMode); err != nil {
		return fmt.Errorf("policy contains invalid variables: %s", err.Error())
	}

	if backgroundMode {
		if err := containsUserVariables(p, vars); err != nil {
			return fmt.Errorf("only select variables are allowed in background mode. Set spec.background=false to disable background mode for this policy rule: %s ", err)
		}
	}

	return nil
}

```